changefeedtrigger/__init__.py

```diff
@@ -0,0 +1,94 @@
+    if blob_name in tokens:
+        cursor = filter_event_service.download_blob(blob_name).readall()
+        # if value of token is None return None
+        if cursor == b"None":
+            return None
+        # if value of token is a cursor initiate the changefeed with cursor
+        elif cursor != b"" and cursor != b"None":
+            change_feed = cf_client.list_changes(results_per_page=500).by_page(continuation_token=eval(cursor))
+        # if token is empty initiate changefeed without cursor
+        else:
+            change_feed = cf_client.list_changes(results_per_page=500).by_page()
+    # if token does not exist upload token as empty string placeholder as blob and initiate changefeed without cursor
+    else:
+        filter_event_service.upload_blob(blob_name, "") 
+        change_feed = cf_client.list_changes(results_per_page=500).by_page()
+
+    # iterate through change feed
+    change_feed_page = next(change_feed)
+
+    # get multidimensionally filtered events
+    #filtered_events = list(filter(where(data__eventType__iexact=e_filter, data__subject__icontains=("/" + c_filter + "/blobs/" + b_filter)), events))
+    filtered_events = []
+    for event in change_feed_page:
+        if (event["eventType"] == e_filter) and (("/" + c_filter + "/") in event["subject"]) and (("/" + b_filter) in event["subject"]):
+            filtered_events.append(event)
+        # set continuation token as last event in the filtered event list
+        cursor = change_feed.continuation_token
+    
+    # update continuation token blob with new continuation token
+    filter_event_service.upload_blob(blob_name, str(cursor), overwrite=True)
+
+    # return filtered events
+    return filtered_events
+    
+
+def main(mytimer: func.TimerRequest) -> None:
+    # creating clients from storage account connection string
+    connection_strs = ""
+
+    # create blob service client to access connection string
+    bs_client = BlobServiceClient.from_connection_string(conn_str=connection_strs)
+    change_feed_client = ChangeFeedClient.from_connection_string(conn_str=connection_strs)
+
+    # create or get container to store cursor
+    container_name = "cursor-container"
+    try:
+        cont_client = bs_client.create_container(container_name)
+    except:
+        cont_client = bs_client.get_container_client(container_name)
+
+    # these are the filters being used to filter out specific events
+    # these values represent what event characteristics you want to see
+    # for example, these filters will only return events with the event
+    # type of BlobCreated, and are in the container 
+    # testing-changefeed-container, and that are named blob2
+    event_filter = "BlobCreated"
+    container_filter = "testing-changefeed-container"
+    blob_filter = "blob2"
+
+    # filtering method
+    events = multidimensional_filtering(change_feed_client, blob_filter, bs_client, container_name, container_filter, event_filter)
```

Comment:
I think `multidimensional_filtering` should be named like `get/fetch_events` and should take a predicate as parameter. The predicate can be a named function called for example `event_matches` or so.

See some examples how this could be done:
https://www.askpython.com/python/built-in-methods/python-filter-function
https://stackoverflow.com/questions/290424/filter-a-python-list-by-predicate